src/node_buffer.cc

```diff
@@ -580,7 +582,6 @@ void DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
 
   ArrayBufferViewContents<char> buffer(args[0]);
 
-  CHECK(args[1]->IsBoolean());
```

Comment:
There is only 1 call to decodeUTF8() inside lib/:
https://github.com/nodejs/node/blob/8d0c214a8f0d0742be2b6f04b18810e29a80e1a3/lib/internal/encoding.js#L433
and this[kIgnoreBOM] is already a boolean. How would turning this into an optional parameter help?

Reply:
No need to check for boolean, since isTrue() is enough.